airbyte-webapp-e2e-tests/cypress/integration/connection.spec.ts

```diff
@@ -121,6 +127,123 @@ describe("Connection main actions", () => {
+    checkSuccessResult();
+
+    deleteSource(sourceName);
+    deleteDestination(destName);
+  });
+
+  // Analytics calls are not made in CI.
+  it.skip("Should make analytics calls", () => {
+    const sourceName = appendRandomString("Test connection source cypress PokeAPI");
+    const destName = appendRandomString("Test connection destination cypress");
+
+    createTestConnection(sourceName, destName);
+
+    cy.get("div").contains(sourceName).should("exist");
+    cy.get("div").contains(destName).should("exist");
+
+    openSourceDestinationFromGrid(sourceName);
+
+    goToReplicationTab();
+
+    cy.intercept("https://api.segment.io/v1/t").as("analyticsCall");
+
+    let analyticsCalls = 0;
+
+    selectSchedule("Cron");
+    cy.wait("@analyticsCall").then(() => {
+      analyticsCalls++;
+    });
+    submitButtonClick();
+    checkSuccessResult();
+
+    selectSchedule("Manual");
+    cy.wait("@analyticsCall").then(() => {
+      analyticsCalls++;
+    });
+    submitButtonClick();
+    checkSuccessResult();
+
+    selectSchedule("Every hour");
+    cy.wait("@analyticsCall").then(() => {
+      analyticsCalls++;
+      expect(analyticsCalls).to.eq(3);
+    });
+    submitButtonClick();
+    checkSuccessResult();
+
+    deleteSource(sourceName);
+    deleteDestination(destName);
+  });
+
+  it("Saving a connection's schedule type only changes expected values", () => {
+    cy.intercept("/api/v1/web_backend/connections/update").as("updateConnection");
+    cy.intercept("/api/v1/web_backend/connections/get").as("getConnection");
+
+    const sourceName = appendRandomString("Test update connection PokeAPI source cypress");
+    const destName = appendRandomString("Test update connection Local JSON destination cypress");
+
+    createTestConnection(sourceName, destName);
+
+    goToSourcePage();
+    openSourceDestinationFromGrid(sourceName);
+    openSourceDestinationFromGrid(`${sourceName} <> ${destName}`);
+
+    let loadedConnection: any = null; // Should be a WebBackendConnectionRead
+    cy.wait("@getConnection").then((interception) => {
+      loadedConnection = interception.response?.body;
+      expect(loadedConnection).not.to.eq(null);
+    });
+
+    goToReplicationTab();
+
+    selectSchedule("Every hour");
+    submitButtonClick();
+
+    cy.wait("@updateConnection").then((interception) => {
+      // Schedule is pulled out here, but we don't do anything with is as it's legacy
+      const { scheduleType, scheduleData, schedule, ...connectionUpdate } = interception.response?.body;
+      expect(scheduleType).to.eq("basic");
+      expect(scheduleData.basicSchedule).to.deep.eq({
+        timeUnit: "hours",
+        units: 1,
+      });
+
+      const { scheduleType: readScheduleType, scheduleData: readScheduleData, ...connectionRead } = loadedConnection;
+      expect(readScheduleType).to.eq("manual");
+      expect(readScheduleData).to.eq(undefined);
+      expect(connectionRead).to.deep.eq(connectionUpdate);
```

Comment:
Why not put these checks in the `cy.wait("@getConnection").then((interception) => {})` block? It feels like they conceptually fit there more